src/drop.ts

```diff
@@ -52,3 +58,9 @@ function lazyImplementation<T>(n: number): LazyEvaluator<T> {
     return { done: false, hasNext: true, next: value };
   };
 }
+
+// `ArraySlice<[1, 2, 3], number>` returns `never`, so we handle
+// this case by returning `Array<1 | 2 | 3>` instead.
+type Drop<T extends Array<unknown>, N extends number> = number extends N
```

Comment:
Use our IterableContainer
```suggestion
type Drop<T extends IterableContainer, N extends number> = number extends N
```

Reply:
Done